core/src/main/java/org/web3j/tx/Contract.java

```diff
@@ -393,27 +395,63 @@ TransactionReceipt executeTransaction(
                 ContractEIP1559GasProvider eip1559GasProvider =
                         (ContractEIP1559GasProvider) gasProvider;
                 if (eip1559GasProvider.isEIP1559Enabled()) {
+                    Transaction tx;
+                    if (constructor) {
+                        tx =
+                                Transaction.createContractTransaction(
+                                        this.transactionManager.getFromAddress(),
+                                        BigInteger.ONE,
+                                        gasProvider.getGasPrice(),
+                                        data);
+                    } else {
+                        tx =
+                                Transaction.createFunctionCallTransaction(
+                                        this.transactionManager.getFromAddress(),
+                                        BigInteger.ONE,
+                                        gasProvider.getGasPrice(),
+                                        gasProvider.getGasLimit(),
+                                        contractAddress,
+                                        data);
+                    }
                     receipt =
                             sendEIP1559(
                                     eip1559GasProvider.getChainId(),
                                     contractAddress,
                                     data,
                                     weiValue,
-                                    eip1559GasProvider.getGasLimit(funcName),
-                                    eip1559GasProvider.getMaxPriorityFeePerGas(funcName),
-                                    eip1559GasProvider.getMaxFeePerGas(funcName),
+                                    eip1559GasProvider.getGasLimit(tx),
+                                    eip1559GasProvider.getMaxPriorityFeePerGas(),
+                                    eip1559GasProvider.getMaxFeePerGas(),
                                     constructor);
                 }
             }
 
             if (receipt == null) {
+                Transaction tx;
+                if (constructor) {
+                    tx =
+                            Transaction.createContractTransaction(
+                                    this.transactionManager.getFromAddress(),
+                                    BigInteger.ONE,
+                                    gasProvider.getGasPrice(),
+                                    data);
+                } else {
+                    tx =
+                            Transaction.createFunctionCallTransaction(
+                                    this.transactionManager.getFromAddress(),
+                                    BigInteger.ONE,
+                                    gasProvider.getGasPrice(),
+                                    gasProvider.getGasLimit(),
+                                    contractAddress,
+                                    data);
+                }
```

Comment:
this is duplicated code, should be exported to a method for a better maintaining

Reply:
Done